android/src/main/java/com/reactnativecommunity/webview/RNCWebViewModule.java

```diff
@@ -151,12 +155,42 @@ public void onShouldStartLoadWithRequestCallback(final boolean shouldStart, fina
 
   @ReactMethod
   public void releaseWebView(final String webViewKey) {
-    // no-op for now
+    UiThreadUtil.runOnUiThread(() -> {
+      RNCWebView rncWebView = RNCWebViewMapManager.INSTANCE.getRncWebViewMap().get(webViewKey);
+      RNCWebViewManager.InternalWebView webView = (RNCWebViewManager.InternalWebView) RNCWebViewMapManager.INSTANCE.getInternalWebViewMap().get(webViewKey);
+
+      if (webView == null) {
+        FLog.w(TAG, "Failed to release webview with webViewKey: " + webViewKey);
+        return;
+      }
+
+      // Detach internal webview from the wrapper RNCWebView
+      if (rncWebView != null) {
+        RNCWebViewManager.InternalWebView internalWebView = rncWebView.detachWebView();
+        if (internalWebView != webView) {
+          throw new IllegalStateException("internalWebViewMap has a mismatched webview with key: " + webViewKey);
+        }
+      }
+
+      // Perform webview cleanup
+      if (webView.webViewKey != null || webView.keepWebViewInstanceAfterUnmount) {
+        ((ThemedReactContext) webView.getContext()).removeLifecycleEventListener(webView);
+        webView.cleanupCallbacksAndDestroy();
+      }
+
+      RNCWebViewMapManager.INSTANCE.getRncWebViewMap().remove(webViewKey);
+      RNCWebViewMapManager.INSTANCE.getInternalWebViewMap().remove(webViewKey);
+    });
   }
 
   @ReactMethod
   public void injectJavaScriptWithWebViewKey(final String webViewKey, final String script) {
-    // no-op for now
+    UiThreadUtil.runOnUiThread(() -> {
+      RNCWebView view = RNCWebViewMapManager.INSTANCE.getRncWebViewMap().get(webViewKey);
+      if (view != null) {
+        view.getWebView().setInjectedJavaScript(script);
```

Comment:
Should we use `ifHasInternalWebView` here? It looks like we already null-checked the `RNCWebView`, but I think we still need to null-check the `InternalWebView`.

Reply:
sorry I wasn't supposed to commit that blob (that was from earlier testing i'll revert those changes)

Reply:
done